algorithmic_efficiency/workloads/librispeech_deepspeech/librispeech_pytorch/model.py

```diff
@@ -0,0 +1,393 @@
+
+    self.lin = nn.LazyLinear(out_features=self.encoder_dim, bias=True)
+
+    self.dropout = nn.Dropout(p=self.input_dropout_rate)
+
+  def forward(self, inputs, input_paddings):
+    output_paddings = input_paddings
+    outputs = inputs[:, None, :, :]
+
+    outputs, output_paddings = self.conv1(outputs, output_paddings)
+    outputs, output_paddings = self.conv2(outputs, output_paddings)
+
+    batch_size, channels, subsampled_lengths, subsampled_dims = outputs.shape
+    outputs = outputs.permute(0, 2, 3, 1).reshape(batch_size,
+                                                  subsampled_lengths,
+                                                  subsampled_dims * channels)
+
+    outputs = self.lin(outputs)
+    outputs = self.dropout(outputs)
+
+    return outputs, output_paddings
+
+
+class Conv2dSubsampling(nn.Module):
+
+  def __init__(self,
+               input_channels: int,
+               output_channels: int,
+               filter_stride: Tuple[int] = (2, 2),
+               padding: str = 'SAME',
+               batch_norm_momentum: float = 0.999,
+               batch_norm_epsilon: float = 0.001):
+    super().__init__()
+
+    self.input_channels = input_channels
+    self.output_channels = output_channels
+    self.filter_stride = filter_stride
+    self.padding = padding
+
+    self.filter_shape = (output_channels, input_channels, 3, 3)
+
+    self.kernel = nn.Parameter(
+        torch.nn.init.xavier_uniform_(torch.empty(*self.filter_shape)))
+    self.bias = nn.Parameter(torch.zeros(output_channels))
+
+    self.bn = BatchNorm(
+        dim=output_channels,
+        batch_norm_momentum=batch_norm_momentum,
+        batch_norm_epsilon=batch_norm_epsilon)
+
+  def get_same_padding(self, input_shape):
+    in_height, in_width = input_shape[2:]
+    stride_height, stride_width = self.filter_stride
+    filter_height, filter_width = 3, 3
+    if in_height % stride_height == 0:
+      pad_along_height = max(filter_height - stride_height, 0)
+    else:
+      pad_along_height = max(filter_height - (in_height % stride_height), 0)
+    if in_width % stride_width == 0:
+      pad_along_width = max(filter_width - stride_width, 0)
+    else:
+      pad_along_width = max(filter_width - (in_width % stride_width), 0)
+    pad_top = pad_along_height // 2
+    pad_bottom = pad_along_height - pad_top
+    pad_left = pad_along_width // 2
+    pad_right = pad_along_width - pad_left
+    return (pad_left, pad_right, pad_top, pad_bottom)
+
+  def forward(self, inputs, paddings):
+    groups = inputs.shape[1] // self.input_channels
+
+    if self.padding == 'SAME':
+      in_ = F.pad(inputs, self.get_same_padding(inputs.shape))
+    else:
+      in_ = inputs
+    outputs = F.conv2d(
+        input=in_,
+        weight=self.kernel,
+        bias=self.bias,
+        stride=self.filter_stride,
+        dilation=(1, 1),
+        groups=groups)
+
+    outputs = self.bn(outputs.permute(0, 2, 3, 1)).permute(0, 3, 1, 2)
```

Comment:
remove, this huge BatchNorm was causing issues in JAX deepspeech model 
you can also confirm that it's not used in conformer Conv2dSubsampling layer

Reply:
Done!